src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/BufferInternal.cs

```diff
@@ -0,0 +1,271 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System {
+
+    // These sources are taken from corclr repo (src\mscorlib\src\System\Buffer.cs)
+    // The reason for this duplication is that System.Runtime.dll 4.0.10 did not expose Buffer.MemoryCopy,
+    // but we need to make this component work with System.Runtime.dll 4.0.10
+    // The methods AreOverlapping and SlowCopyBackwards are not from Buffer.cs. Buffer.cs does an internal CLR call for these.
+    static class BufferInternal
+    {       
+        // This method has different signature for x64 and other platforms and is done for performance reasons.
+        [System.Security.SecurityCritical]
+#if WIN64
```

Comment:
Possibly, but there is so much pointer arithmetic here that I really did not want to touch this code. If you feel strongly about it, I will remove the ifdefs.